unblob/handlers/archive/qnap/qnap_nas.py

```diff
@@ -58,6 +58,7 @@ def is_valid_header(header: Instance) -> bool:
 def _hyperscan_match(
     context: QTSSearchContext, pattern_id: int, offset: int, end: int
 ) -> Scan:
+    del pattern_id, end
```

Comment:
I'd rather use `noqa` than a `del` statement to force use the variable. This is super confusing if you don't know it's related to a linting rule.

Reply:
I've got the idea from vulture https://github.com/jendrikseipp/vulture#marking-unused-variables
Unfortunately `noqa` prevents detecting changes in unused variables.

Another possibility could be to arrign it to `_`:

```suggestion
    _ = pattern_id, end
```

Reply:
Let's keep it if it's industry standard. Maybe just a comment to make it explicit ?

```python
del pattern_id, end # recommended way to mark unused variables according to pylint
```